Bookish.Web/Controllers/HomeController.cs

```diff
@@ -47,6 +48,30 @@ public IActionResult Copies(string isbn)
             return View(model);
         }
 
+        public IActionResult NewBook()
+        {
+            return View();
+        }
+
+        public IActionResult AddBook(string title, string author, string isbn, int totalCopies)
+        {
+            try
+            {
+                bookishService.AddBook(title, author, isbn, totalCopies);
+            }
+            catch
```

Comment:
What kind of exceptions are you expecting to occur? 🙂

Reply:
Mainly with duplicate ISBNs.

Reply:
Ah okay. If it is guaranteed to throw an exception and not leave the db in a difficult state (e.g. if it doesn't add the book but does add copies) then it's fine, although I would expect you to catch a specific exception here. Otherwise you can just have a separate check to see if the isbn is already in use 🙂